presto-orc/src/test/java/com/facebook/presto/orc/stream/AbstractTestLongStreamAriaScan.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.stream;
+
+import com.facebook.presto.orc.OrcCorruptionException;
+import com.facebook.presto.orc.OrcDataSourceId;
+import com.facebook.presto.orc.checkpoint.LongStreamCheckpoint;
+import com.facebook.presto.orc.metadata.Stream;
+import com.facebook.presto.orc.metadata.Stream.StreamKind;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public abstract class AbstractTestLongStreamAriaScan<W extends LongOutputStream, R extends LongInputStream>
```

Comment:
The inheritance in this class is confusing me and I think it may be unnecessary.  Instead of the generics above and the factory methods that use them, can we just have lambdas into a static method `testScan` which is a utility?

Reply:
That is a good point. These tests are a copy of AbstractTestValueStream specialized for LongInputStream subclasses. 
IMO it's less confusing than passing lambdas, given that it is the aria analog of the AbstractTestValueStream test classes but we can talk about this more.
Maybe we can modify both AbstractTestValueStream and AbstractTestLongStreamAriaScan in a separate pull request if we decide that it's less confusing to pass lambdas vs abstract methods.

What do you think about that?

Reply:
Let's start with a simpler point, can you explain why we need the class-level generics `W` and `R`?  It's not clear to me why we can't just have abstract methods that simply return `LongOutputStream` and `LongInputStream`.

Reply:
We can. I originally intended this to be AbstractValueStreamAriaScan, analogous to AbstractValueStream with the intent to have this work for all the aria stream readers.

Then I changed it to AbstractLongStream following @mbasmanova's suggestion in this pr. 

Going forward if this is only used for LongInputStream I can remove the generics and add abstract methods (now leaning towards this).

If this is going to be used for more than LongInputStream then I can go back and make it more generic. If that is the case then I won't put the test data in the abstract class (suggestion above).

How do we all feel about that?

Reply:
@elonazoulay I don't see how it would be generalized to other readers. `scan` method being tested is specific to long reader. Hence, I prefer to simplify the test to make it long-specific.